OsmAndMapCreator/src/net/osmand/data/preparation/IndexPoiCreator.java

```diff
@@ -190,15 +202,25 @@ public void removePoiFile() {
 	}
 
 
-	private void insertAmenityIntoPoi(Amenity amenity) throws SQLException {
+	public boolean insertAmenityIntoPoi(Amenity amenity) throws SQLException {
 		assert IndexConstants.POI_TABLE != null : "use constants here to show table usage "; //$NON-NLS-1$
 		poiPreparedStatement.setLong(1, amenity.getId());
 		poiPreparedStatement.setInt(2, MapUtils.get31TileNumberX(amenity.getLocation().getLongitude()));
 		poiPreparedStatement.setInt(3, MapUtils.get31TileNumberY(amenity.getLocation().getLatitude()));
 		poiPreparedStatement.setString(4, amenity.getType().getKeyName());
 		poiPreparedStatement.setString(5, amenity.getSubType());
 		poiPreparedStatement.setString(6, encodeAdditionalInfo(amenity, amenity.getAdditionalInfo(), amenity.getName(), amenity.getEnName(false)));
-		addBatch(poiPreparedStatement);
+		try {
+			addBatch(poiPreparedStatement);
+			return true;
+		} catch (SQLException e) {
+			if (e.getMessage().contains("UNIQUE constraint failed: ")) {
```

Comment:
This should not be here. We should not catch exception this is a very bad practice.